Source/Core/Core/PatchEngine.cpp

```diff
@@ -211,6 +224,52 @@ static void ApplyPatches(const std::vector<Patch>& patches)
   }
 }
 
+std::unique_ptr<std::map<u64, std::vector<u8>>> CalculateDiscPatches(const DiscIO::Volume &disc) {
+  if (s_file_patches.empty())
```

Comment:
Seems like an extra unneeded check? Why not return a size 0 collection? That way you can also remove the nullptr check in users.